### PR TITLE
Bug fix: point mechs applying weights to ionic concentrations

### LINKED ISSUE #1848: Writing ionic concentrations in point mechanisms
```
NEURON {
    POINT_PROCESS syn
    NONSPECIFIC_CURRENT i
    USEION p WRITE pi WRITE po
}

ASSIGNED {}

INITIAL {
        pi = 1
}

STATE {
    pi
}

BREAKPOINT {
    SOLVE state METHOD cnexp
}

DERIVATIVE state {
    pi' = -pi/2
}

```
Produces the following functions when compiled with modcc for CPU (unvectorized): 
```
static void write_ions(arb_mechanism_ppack* pp) {
    PPACK_IFACE_BLOCK;
    for (arb_size_type i_ = 0; i_ < _pp_var_width; ++i_) {
        auto ion_p_indexi_ = _pp_var_ion_p_index[i_];
        arb_value_type pi_shadowed_ = 0;
        pi_shadowed_ = _pp_var_pi[i_];
        _pp_var_ion_p.internal_concentration[ion_p_indexi_] = fma(_pp_var_weight[i_], pi_shadowed_, _pp_var_ion_p.internal_concentration[ion_p_indexi_]);
    }
}
```
The problem is in the update to the internal concentration. We scale `pi_shadowed_` by a weight variable that is equal to 1000/cv_area (see [here](https://github.com/arbor-sim/arbor/blob/master/arbor/fvm_lowered_cell_impl.hpp#L551)). What this means is that the effective unit of the concentration in the nmodl file is (um^2)*Kmol/L. 
The weight factor was calculated for the current contribution, and should not be included in the calculation of ioninc concentrations.

### PR DESCRIPTION
**Pull request by thorstenhater**
Mechanisms of the `POINT_PROCESS` variety no longer apply `weight` to concentrations `Xo`/`Xi`

Fixes #1848

### CODE REVIEW

**Comment by thorstenhater** [comment]
bors try

**Comment by thorstenhater** [comment]
⚠️ This seems to change the number of spikes generated by the ring example ⚠️

File: modcc/printer/cprinter.cpp [hunk 1]
```diff
@@ -561,70 +564,65 @@ std::list<index_prop> gather_indexed_vars(const std::vector<LocalVariable*>& ind
     return indices;
 }
 
-void emit_state_read(std::ostream& out, LocalVariable* local) {
-    ENTER(out);
```
Comment (boeschf) [inline modcc/printer/cprinter.cpp:565]: you deleted these debug statements, was that on purpose?

File: modcc/printer/gpuprinter.cpp [hunk 2]
```diff
@@ -429,8 +429,8 @@ void emit_api_body_cu(std::ostream& out, APIMethod* e, bool is_point_proc, bool
                 out << "unsigned lane_mask_ = arb::gpu::ballot(0xffffffff, tid_<n_);\n";
             }
         }
-        ppack && out << "PPACK_IFACE_BLOCK;\n";
-        cv_loop && out << "if (tid_<n_) {\n" << indent;
+        flags.ppack_iface && out << "PPACK_IFACE_BLOCK;\n";
+        flags.cv_loop && out << "if (tid_<n_) {\n" << indent;
```
Comment (boeschf) [inline modcc/printer/gpuprinter.cpp:433]: More readable, and inline with your changes in `cprinter`:
```suggestion
        if (flags.ppack_iface) out << "PPACK_IFACE_BLOCK;\n";
        if (flags.cv_loop) out << "if (tid_<n_) {\n" << indent;
```

File: modcc/printer/gpuprinter.cpp [hunk 3]
```diff
@@ -444,9 +444,9 @@ void emit_api_body_cu(std::ostream& out, APIMethod* e, bool is_point_proc, bool
         out << cuprint(body);
 
         for (auto& sym: indexed_vars) {
-            emit_state_update_cu(out, sym, sym->external_variable(), is_point_proc, additive);
+            emit_state_update_cu(out, sym, sym->external_variable(), flags);
         }
-        cv_loop && out << popindent << "}\n";
+        flags.cv_loop && out << popindent << "}\n";
```
Comment (boeschf) [inline modcc/printer/gpuprinter.cpp:449]:
```suggestion
        if (flags.cv_loop) out << popindent << "}\n";
```

**Review by boeschf — APPROVED** [review]

**Pull request merged**